plugins/woocommerce/src/Admin/API/Init.php

```diff
@@ -54,115 +55,167 @@ public function __construct() {
 	}
 
 	/**
-	 * Init REST API.
+	 * Initialize the API namespaces under WooCommerce Admin.
+	 *
+	 * @return void
 	 */
 	public function rest_api_init() {
-		$controllers           = array();
-		$analytics_controllers = array();
-
 		if ( wc_rest_should_load_namespace( 'wc-admin' ) ) {
-			// Controllers in the wc-admin namespace.
-			$controllers = array(
-				'Automattic\WooCommerce\Admin\API\Notice',
-				'Automattic\WooCommerce\Admin\API\Features',
-				'Automattic\WooCommerce\Admin\API\Experiments',
-				'Automattic\WooCommerce\Admin\API\Marketing',
-				'Automattic\WooCommerce\Admin\API\MarketingOverview',
-				'Automattic\WooCommerce\Admin\API\MarketingRecommendations',
-				'Automattic\WooCommerce\Admin\API\MarketingChannels',
-				'Automattic\WooCommerce\Admin\API\MarketingCampaigns',
-				'Automattic\WooCommerce\Admin\API\MarketingCampaignTypes',
-				'Automattic\WooCommerce\Admin\API\Options',
-				'Automattic\WooCommerce\Admin\API\Settings',
-				'Automattic\WooCommerce\Admin\API\PaymentGatewaySuggestions',
-				'Automattic\WooCommerce\Admin\API\Themes',
-				'Automattic\WooCommerce\Admin\API\Plugins',
-				'Automattic\WooCommerce\Admin\API\OnboardingFreeExtensions',
-				'Automattic\WooCommerce\Admin\API\OnboardingProductTypes',
-				'Automattic\WooCommerce\Admin\API\OnboardingProfile',
-				'Automattic\WooCommerce\Admin\API\OnboardingTasks',
-				'Automattic\WooCommerce\Admin\API\OnboardingThemes',
-				'Automattic\WooCommerce\Admin\API\OnboardingPlugins',
-				'Automattic\WooCommerce\Admin\API\OnboardingProducts',
-				'Automattic\WooCommerce\Admin\API\MobileAppMagicLink',
-				'Automattic\WooCommerce\Admin\API\ShippingPartnerSuggestions',
-			);
+			$this->rest_api_init_wc_admin();
+		}
+
+		$rest_api_util = wc_get_container()->get( RestApiUtil::class );
+		if ( is_callable( array( $rest_api_util, 'lazy_load_namespace' ) ) ) {
+			$rest_api_util->lazy_load_namespace( 'wc-analytics', array( $this, 'rest_api_init_wc_analytics' ) );
+		} else {
+			$this->rest_api_init_wc_analytics();
 		}
 
 		if ( Features::is_enabled( 'launch-your-store' ) ) {
-			$controllers[] = 'Automattic\WooCommerce\Admin\API\LaunchYourStore';
+			$controller        = 'Automattic\WooCommerce\Admin\API\LaunchYourStore';
+			$this->$controller = new $controller();
```

Comment:
What is going on here? Why are we assigning this controller to `$this->$controller`? 

Its not referenced anywhere else. Shouldn't we just call  `register_routes`? 

Which I might add is not lazy loaded either. Should it be?

Reply:
> What is going on here? Why are we assigning this controller to $this->$controller?
> Its not referenced anywhere else. Shouldn't we just call register_routes?

This is just maintaining backward compatibility.  I have no idea why these were dynamically assigned to the class, but they were and they're automatically public, so there is risk in removing it.

> Which I might add is not lazy loaded either. Should it be?
No, I'm only lazy loading analytics for this PR.